velox/functions/lib/Re2Functions.cpp

```diff
@@ -817,6 +1055,89 @@ std::shared_ptr<VectorFunction> makeRe2Extract(
   }
 }
 
+std::vector<std::shared_ptr<exec::FunctionSignature>> re2ReplaceSignatures(){
+  return {
+
+    // regex_replace (str, pattern, replace)
+    exec::FunctionSignatureBuilder()
+          .returnType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .build(),
+    // regex_replace (str, pattern, replace, position)
+    exec::FunctionSignatureBuilder()
+          .returnType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .argumentType("integer")
+          .build(),
+        // regex_replace (str, pattern, replace, position)
+    exec::FunctionSignatureBuilder()
+          .returnType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .argumentType("bigint")
+          .build(),
+  };
+}
+
+std::shared_ptr<VectorFunction> makeRe2Replace(
+    const std::string& name,
+    const std::vector<VectorFunctionArg>& inputArgs,
+    const core::QueryConfig& /*config*/
+    ) {
+
+  auto numArgs = inputArgs.size();
+
+  VELOX_USER_CHECK(
+    numArgs == 3 || numArgs == 4,
+    "{} requires between 2 and 4 arguments, but got {}",
+    name,
+    numArgs);
+
+  VELOX_USER_CHECK(
+      inputArgs[0].type->isVarchar(),
+      "{} requires the first argument of type VARCHAR, but got {}",
+      name,
+      inputArgs[0].type->toString());
+
+  VELOX_USER_CHECK(
+      inputArgs[1].type->isVarchar(),
+      "{} requires the second argument of type VARCHAR, but got {}",
+      name,
+      inputArgs[1].type->toString());
+
+  BaseVector* constantPattern = inputArgs[1].constantValue.get();
+
+  VELOX_USER_CHECK(
+    inputArgs[2].type->isVarchar(),
+    "{} requires the thrid argument of type VARCHAR, but got {}",
+    name,
+    inputArgs[1].type->toString());
+
+  // Check for the 4th argument if necessary
+  if (numArgs == 4){
+    VELOX_USER_CHECK(
+      inputArgs[3].type->isInteger() || inputArgs[3].type->isBigint(),
+      "{} requires the fourth argument of type INTEGER, but got {}",
+      name,
+      inputArgs[3].type->toString());
+  }
+
+  if (constantPattern != nullptr && !constantPattern->isNullAt(0)) {
+    auto pattern = constantPattern->as<ConstantVector<StringView>>()->valueAt(0);
+    // Use the overwrite value and logic as needed
+    return std::make_shared<Re2ReplaceConstantPattern<int64_t>>(pattern);
```

Comment:
Should also add logic `Re2ReplaceConstantPattern<int32_t>` based on your signatures definition.

Reply:
Removed <int32_t> and all templates in the next commit. Functions are not dependant on type passed, output is always a string